src/main/java/leo/storage/Storage.java

```diff
@@ -0,0 +1,238 @@
+package leo.storage;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import leo.leoexception.IncorrectMarkException;
+import leo.leoexception.LeoException;
+import leo.leoexception.NoStorageFileException;
+import leo.leoexception.NoTaskFoundException;
+import leo.ui.Ui;
+
+/**
+ * Represents a storage place containing a list Tasks. A <code>Storage</code> object corresponds to
+ * a database consisting of the list of Tasks, the File in which the data is stored in and the file path.
+ */
+public class Storage {
+
+    private final TaskList data;
+    private final String dataFilePath;
+    private final File taskFile;
+
+    /**
+     * Constructor for creating a Storage object.
+     *
+     * @param filePath The path where the Tasks are stored.
+     * @throws NoStorageFileException If file cannot be found.
+     * @throws IncorrectMarkException If Task has already been marked.
+     */
+    public Storage(String filePath) throws NoStorageFileException, IncorrectMarkException {
+        String root = Paths.get("").toAbsolutePath().toString();
+        this.dataFilePath = Paths.get(root, filePath).toString();
+
+        this.taskFile = new File(this.dataFilePath);
+        this.data = new TaskList(loadData());
+    }
+
+    /**
+     * Returns a list of tasks stored in the data file.
+     *
+     * @return List of tasks.
+     * @throws IncorrectMarkException If a completed task is set to be completed.
+     * @throws NoStorageFileException If file is not found.
+     */
+    private List<Task> loadData() throws IncorrectMarkException, NoStorageFileException {
+        List<Task> taskList = new ArrayList<>();
+        if (!taskFile.exists()) {
+            return taskList;
+        }
+
+        try {
+            Scanner scanner = new Scanner(taskFile);
+            while (scanner.hasNextLine()) {
+                String task = scanner.nextLine().strip();
+                char taskType = task.charAt(1);
+                char completion = task.charAt(4);
+                String description = task.substring(7).strip();
+
+                try {
+                    if (taskType == 'T') {
+                        Task t = new ToDoTask(description);
+                        if (completion == 'X') {
+                            t.mark();
+                        }
+                        taskList.add(t);
+                    } else {
+                        String[] temp = description.split("\\|");
+                        String taskDescription = temp[0].strip();
+                        String time = temp[1].strip();
+                        LocalDateTime dt = convertString(time);
```

Comment:
Very very small detail but maybe make the variable name more clear? dateTime instead of dt?